File: complete/3273.cpp
#include <algorithm>
#include <iostream>

int main() {
  std::cin.tie(NULL);
  std::cout.tie(NULL);
  std::ios_base::sync_with_stdio(false);

  int n;
  std::cin >> n;
  int *arr;
  arr = new int[n];
  for (int i = 0; i < n; i++) {
    std::cin >> arr[i];
  }
  int sum;
  std::cin >> sum;

  std::sort(arr, arr + n);

  int *p1, *p2;

  p1 = arr;
  p2 = arr + n - 1;
  int count = 0;

  while (p1 < p2) {
    int current_sum = *p1 + *p2;
    if (current_sum == sum) {
      p1++;
      count++;
    } else if (current_sum > sum) {
      p2--;
    } else {
      p1++;
    }
  }
  std::cout << count;
  return 0;
}

File: complete/3273.py
n = int(input())
lst = map(int, input().split(' '))
Sum = int(input())
count = 0

lst.sort()
for i in range(len(lst) - 1):
    for j in range(i + 1, len(lst)):
        current_sum = lst[i] + lst[j]
        if Sum == current_sum:
            count += 1
        elif Sum < current_sum:
            break
print(count)